[kube-prometheus-stack] Default values prevent to override operator configuration

### Describe the bug a clear and concise description of what the bug is.

At differente places we can see these config

https://github.com/prometheus-community/helm-charts/blob/b320323b9852a169145b3aa64e98af7eae3a0178/charts/kube-prometheus-stack/values.yaml#L706C1-L708C44

```
    ## Namespaces to be selected for AlertmanagerConfig discovery. If nil, only check own namespace.
    ##
    alertmanagerConfigNamespaceSelector: {}
```

Thanks to this comment https://github.com/prometheus-operator/prometheus-operator/issues/5327#issuecomment-1424036851

> I see, then if I'm not mistaken you also have to set alertmanagerConfigNamespaceSelector to {} to select all the namespaces, if it's nil (which is the default if you don't configure it) the operator will look for AlertmanagerConfig in the other namespaces.

The helm chart forces the operator to look across all namespaces. One could ask if the helm chart should follow the default values of the operator or if having opinionated values is better, but that's another question.

The main issue here is that if you want to override the value by nil/null, helm will raise an error because types can't be merged

```
coalesce.go:286: warning: cannot overwrite table with non table for prometheus-stack.kube-prometheus-stack.alertmanager.alertmanagerSpec.alertmanagerConfigNamespaceSelector (map[])
```

### What's your helm version?

version.BuildInfo{Version:"v3.15.2", GitCommit:"1a500d5625419a524fdae4b33de351cc4f58ec35", GitTreeState:"clean", GoVersion:"go1.22.4"}

### What's your kubectl version?

Client Version: v1.29.6 Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3 Server Version: v1.30.1

### Which chart?

kube-prometheus-stack

### What's the chart version?

61.2.0

### What happened?

nil value can't be merged with a map

### What you expected to happen?

I need a solution which allow me to set null or map like the operator allows

### How to reproduce it?

run helm template of the chart with this override

```
  alertmanager:
    alertmanagerSpec:
      alertmanagerConfigNamespaceSelector:
```      

### Enter the changed values of values.yaml?

_No response_

### Enter the command that you execute and failing/misfunctioning.

helm template -f prometheus-stack/values.yaml charts/prometheus-stack | more

### Anything else we need to know?

I was working on the alertmanagerConfigNamespaceSelector value but there is different other places where this bug happens.

I think the best way to fix it would be to have a default value to nil. It can be override by a map and it also align with the default values of the operator

The bug is maybe only related to helm https://github.com/helm/helm/pull/12879

--- a/charts/kube-prometheus-stack/Chart.yaml
+++ b/charts/kube-prometheus-stack/Chart.yaml
@@ -31,7 +31,7 @@ name: kube-prometheus-stack
 sources:
   - https://github.com/prometheus-community/helm-charts
   - https://github.com/prometheus-operator/kube-prometheus
-version: 75.3.0
+version: 75.3.1
 
 # Please do not add a renovate hint here, since appVersion updates involves manual tasks
 appVersion: v0.83.0

--- a/charts/kube-prometheus-stack/Chart.yaml
+++ b/charts/kube-prometheus-stack/Chart.yaml
@@ -31,7 +31,7 @@ name: kube-prometheus-stack
 sources:
   - https://github.com/prometheus-community/helm-charts
   - https://github.com/prometheus-operator/kube-prometheus
-version: 75.3.1
+version: 75.3.2
 
 # Please do not add a renovate hint here, since appVersion updates involves manual tasks
 appVersion: v0.83.0

--- a/charts/kube-prometheus-stack/templates/alertmanager/alertmanager.yaml
+++ b/charts/kube-prometheus-stack/templates/alertmanager/alertmanager.yaml
@@ -75,12 +75,8 @@ spec:
 {{ else }}
   alertmanagerConfigSelector: {}
 {{- end }}
-{{- if .Values.alertmanager.alertmanagerSpec.alertmanagerConfigNamespaceSelector }}
   alertmanagerConfigNamespaceSelector:
-{{ tpl (toYaml .Values.alertmanager.alertmanagerSpec.alertmanagerConfigNamespaceSelector | indent 4) . }}
-{{ else }}
-  alertmanagerConfigNamespaceSelector: {}
-{{- end }}
+{{ tpl (toYaml .Values.alertmanager.alertmanagerSpec.alertmanagerConfigNamespaceSelector | indent 4 | default "null") .  }}
 {{- if .Values.alertmanager.alertmanagerSpec.web }}
   web:
 {{ toYaml .Values.alertmanager.alertmanagerSpec.web | indent 4 }}

--- a/charts/kube-prometheus-stack/unittests/alertmanager/alertmanager_test.yaml
+++ b/charts/kube-prometheus-stack/unittests/alertmanager/alertmanager_test.yaml
@@ -24,3 +24,26 @@ tests:
       - equal:
           path: spec.tolerations[0].key
           value: key1
+  # 5797
+  - it: should be empty value if alertmanagerConfigNamespaceSelector is not set
+    set:
+      alertmanager.alertmanagerSpec.alertmanagerConfigNamespaceSelector: 
+    asserts:
+      - hasDocuments:
+          count: 1
+      - equal:
+          path: spec.alertmanagerConfigNamespaceSelector
+          value: null
+
+  - it: should be empty map object if alertmanagerConfigNamespaceSelector is empty map
+    set:
+      alertmanager.alertmanagerSpec.alertmanagerConfigNamespaceSelector: {}
+    asserts:
+      - hasDocuments:
+          count: 1
+      - equal:
+          path: spec.alertmanagerConfigNamespaceSelector
+          value: {}
+
+
+    

